src/smtencoding/FStar.SMTEncoding.Encode.fs

```diff
@@ -817,8 +804,12 @@ let encode_top_level_let :
                     | Tm_fvar fv when S.fv_eq_lid fv FStar.Parser.Const.logical_lid -> true
                     | _ -> false
                   in
-                  let is_prims = lbn |> FStar.Compiler.Util.right |> lid_of_fv |> (fun lid -> lid_equals (lid_of_ids (ns_of_lid lid)) Const.prims_lid) in
-                  if not is_prims && (quals |> List.contains Logic || is_logical)
+                  let is_smt_theory_symbol =
+                      let fv = FStar.Compiler.Util.right lbn in
+                      Env.fv_has_attr env.tcenv fv FStar.Parser.Const.smt_theory_symbol_attr_lid
+                  in
+                  if not is_smt_theory_symbol
```

Comment:
It used to be that the logical encoding of definitions with type `logical` was disabled for all symbols in prims. 
I think this was a mistake. It meant that things like

```let ( === ) a b x y == a==b /\ x==y``

were getting encoded via the deep embedding `l_and (...) (...)` as opposed to `(and (= a b) (= x y))`. This also meant that other symbols in prims, e.g., `subtype_of` was getting encoded via the deep-embedding (`l_forall` etc) instead of the more efficient shallow embedding.

I've changed this now to say that a definition inferred to be logical gets the logical encoding (regardless of prims) except if it has the smt_theory_symbol attribute (e.g., the encoding of `/\` etc.), since that already gets special treatment